src/api.js

```diff
@@ -782,6 +782,13 @@ API.prototype.declare = function(options, handler) {
     options.output = this._options.schemaPrefix + options.output;
   }
   this._entries.push(options);
+  var valueArr = options.map((item) => {return item.name;});
+  var isDuplicate = valueArr.some(function(item, idx) { 
```

Comment:
This seems to be searching for a duplicate entry in `valueArr`, which is an array of .. well, I'm not sure what map returns when run over an object.  I think what you want here is `this._entries`, which has the list of entries that have already been defined.

Reply:
in that case, where will I check for options.name? Would it be var isDuplicate = this._entries.some(function(options.name, idx) ?

Reply:
You'll check that below, as I suggested.